src/services/cgnService.ts

```diff
@@ -127,4 +128,40 @@ export default class CgnService {
         }
       });
     });
+
+  /**
+   * Get the current CGN Activation status related to the user.
+   */
+  public readonly getCgnActivation = (
+    user: User
+  ): Promise<
+    // tslint:disable-next-line: max-union-size
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseSuccessJson<CgnActivationDetail>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.cgnApiClient.getCgnActivation({
+        fiscalcode: user.fiscal_code
+      });
+
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 401:
+            return ResponseErrorUnexpectedAuthProblem();
+          case 404:
+            return ResponseErrorNotFound(
+              "Not Found",
+              "No User CGN activation found"
+            );
+          case 500:
+            return ResponseErrorInternal(readableProblem(response.value));
+          default:
+            return ResponseErrorStatusNotDefinedInSpec(response);
```

Comment:
The definition of this function is duplicated in this file and into `src/services/bonusService.ts`